packages/blocks/src/api/parser/convert-legacy-block.js

```diff
@@ -88,25 +88,37 @@ export function convertLegacyBlockNameAndAttributes( name, attributes ) {
 			( name === 'core/paragraph' ||
 				name === 'core/heading' ||
 				name === 'core/image' ||
-				name === 'core/button' )
+				name === 'core/button' ) &&
+			newAttributes.metadata.bindings.__default?.source !==
+				'core/pattern-overrides'
 		) {
 			const bindings = [
 				'content',
 				'url',
 				'title',
+				'id',
 				'alt',
 				'text',
 				'linkTarget',
 			];
+			// Delete any existing individual bindings and add a default binding.
+			// It was only possible to add all the default attributes through the UI,
+			// So as soon as we find an attribute, we can assume all default attributes are overridable.
+			let hasPatternOverrides = false;
 			bindings.forEach( ( binding ) => {
 				if (
-					newAttributes.metadata.bindings[ binding ]?.source?.name ===
-					'pattern_attributes'
+					newAttributes.metadata.bindings[ binding ]?.source ===
+					'core/pattern-overrides'
 				) {
-					newAttributes.metadata.bindings[ binding ].source =
-						'core/pattern-overrides';
+					hasPatternOverrides = true;
+					delete newAttributes.metadata.bindings[ binding ];
 				}
 			} );
+			if ( hasPatternOverrides ) {
+				newAttributes.metadata.bindings.__default = {
+					source: 'core/pattern-overrides',
+				};
+			}
```

Comment:
This works ok in the site editor, but for some reason isn't working in the post editor when navigating to the original pattern. It must be that `convertLegacyBlockNameAndAttributes` is never called when switching to editing a different entity.

Reply:
> This works ok in the site editor, but for some reason isn't working in the post editor when navigating to the original pattern.

Could you clarify the testing instructions for this? I'm seeing a consistent result on my end.

https://github.com/WordPress/gutenberg/assets/7753001/cb09f8fc-31ee-41f0-8549-879dfba6895e

It won't automatically change from `content` to `__default` as shown in the recording, but I think it's expected as the function only runs on change. It's also backward compatible this way, I believe. Maybe there's something I'm missing though 🤔.